### DISCUSSION TITLE
Q4 blog post

### DISCUSSION THREAD

**Discussion by MabezDev** [MabezDev/blog#17] [Ideas]
Title :)

**Comment by bjoernQ** [comment]
## esp-wifi
- WiFi support for ESP32-C2
- basic WiFi power safe support
- BTLE for ESP32-S3

## esp-hal
- ESP32-C2 support
- I2S TX and RX
- DMA (SPI and I2S)
- USB device support

**Comment by SergioGasquez** [comment]
Not sure if worth mentioning in the blog post, so feel free to ignore both suggestions:
- `espup` had a lot of improvements, including installation now being async, and supporting ESP32-C2, and it's now used in the `xtensa-toolchain` action.
- For wokwi.com/rust was updated and now supports `Cargo.toml` for `no_std` projects.

I'm sure you already have it tracked but we should mention that the first 10 patches have been accepted in upstream LLVM.
